build-advisories/src/Roave/SecurityAdvisories/VersionConstraint.php

```diff
@@ -0,0 +1,207 @@
+<?php
+
+namespace Roave\SecurityAdvisories;
+
+/**
+ * A simple version constraint - naively assumes that it is only about ranges like ">=1.2.3,<4.5.6"
+ */
+final class VersionConstraint
+{
+    const CLOSED_RANGE_MATCHER    = '^>(=?)\s*((\d+.)*\d+)\s*,\s*<(=?)\s*((\d+.)*\d+)$';
```

Comment:
you should use non-capturing groups for the ``(\d+.)*``.


And the dot needs escaping, otherwise it matches anything, not just a dot.
And once this is done, you can change all your quantifiers to possessive ones to optimize the regex by avoiding useless backtracking (the character following a repetition can never be matched in the repetition, so backtracking will not change anything)